tests/e2e/models/CMakeLists.txt

```diff
@@ -85,3 +85,61 @@ iree_check_single_backend_test_suite(
 )
 
 ### BAZEL_TO_CMAKE_PRESERVES_ALL_CONTENT_BELOW_THIS_LINE ###
+
+iree_static_linker_test(
+  NAME
+    mnist_fake_weights_llvm_cpu_static_bytecode_test
```

Comment:
When I add new tests, I like to break them so I know that they're working and produce useful logs. Can you try that here? Maybe revert https://github.com/iree-org/iree/pull/10096 locally and share the output of configuring/building/running these tests?

Reply:
Sure. I did test multiple failure cases when I wrote the function. 
If I set the `FUNCTION_INPUTS` for `simple_mul` to `4xf32,f32` then

```
test 15
    Start 15: iree/tests/e2e/models/simple_mul_llvm_cpu_static_c_test

15: Test command: .../tests/e2e/models/simple_mul_llvm_cpu_static_c_test_run "--device=local-sync"
15: Environment variables: 
15:  TEST_TMPDIR=.../test_tmpdir/iree/tests/e2e/models/simple_mul_llvm_cpu_static_c_test_test_tmpdir
15: Test timeout computed to be: 0
15: simple_mul_llvm_cpu_static_c_test encountered error:
15: iree/runtime/src/iree/modules/hal/utils/buffer_diagnostics.c:220: INVALID_ARGUMENT; tensor shape dimension 0 mismatch; expected 4 but have 1; expected shape 4, actual shape 1; while invoking native function hal.buffer_view.assert; while invoking native function module.simple_mul; 
15: [ 1]   native hal.buffer_view.assert:0 -
15: [ 0]   native module.simple_mul:0 -
4/4 Test #15: iree/tests/e2e/models/simple_mul_llvm_cpu_static_c_test ..................***Failed    0.01 sec
```

Whereas the static library name mismatch results in 
```
.../tests/e2e/models/simple_mul_llvm_cpu_static_c_test.c:66:7: error: use of undeclared identifier 'simple_mul_dispatch_library_query'; did you mean 'simple_mul_dispatch_0_library_query'?
      IREE_STATIC_LIB_QUERY_FN,
      ^~~~~~~~~~~~~~~~~~~~~~~~
      simple_mul_dispatch_0_library_query
.../tests/e2e/models/simple_mul_llvm_cpu_static_c_test.c:8:34: note: expanded from macro 'IREE_STATIC_LIB_QUERY_FN'
#define IREE_STATIC_LIB_QUERY_FN simple_mul_dispatch_library_query
                                 ^
.../tests/e2e/models/simple_mul_llvm_cpu_static_c_test.h:21:1: note: 'simple_mul_dispatch_0_library_query' declared here
simple_mul_dispatch_0_library_query(
^
1 error generated.
```